### PR TITLE
feat: updates ban and ban handler

### PR DESCRIPTION
**Pull request by benw202**
## Description

The ban command and ban handler have been refined and a DM has been added to notify the user they have been banned and provides a link to the ban appeal form.

- [x] Ban.ts
- [x] Ban Handler

Notes:
Add reminder to ban handler
Ban handler needs refining
Change link to appeal form in ban.ts

## Test Results

Non bot ban + reminder:
image

Bot ban:
image

Bot ban but has DM's closed:
(`Warn - DM not sent` changed to `Ban - DM not sent`)

image

User DM:
image

## Discord Username

BenW#8484

### CODE REVIEW

File: src/commands/moderation/ban.ts [hunk 1]
```diff
@@ -1,120 +1,154 @@
...
+            name: 'Member',
+            value: user.toString(),
+        },
+        {
+            name: 'Moderator',
+            value: moderator.toString(),
+        },
+        {
+            name: 'Reason',
+            value: `\u200B${reason}`,
```
Comment (NathanInnes) [inline src/commands/moderation/ban.ts:24]: Why zero space character?

File: src/commands/moderation/ban.ts [hunk 2]
```diff
@@ -1,120 +1,154 @@
...
+        {
+            inline: false,
+            name: 'Date',
+            value: formattedDate,
+        },
+        {
+            inline: false,
+            name: 'Appeal',
+            //value: `If you would like to appeal your ban, please fill out [this form.](${process.env.BAN_APPEAL_URL})`,
+            value: 'If you would like to appeal your ban, please fill out [this form.](https://www.google.co.uk/)',
```
Comment (NathanInnes) [inline src/commands/moderation/ban.ts:106]: Is this actually an official form or could you just say to appeal it contact the moderator who banned?

File: src/handlers/ban.ts [hunk 3]
```diff
@@ -12,81 +12,83 @@ module.exports = {
...
-                            value: 'Unavailable - Audit log incomplete',
-                        },
-                        {
-                            name: 'Reason',
-                            value: 'Unavailable - Audit log incomplete',
-                        },
-                    ],
-                    footer: { text: `User ID: ${msg.user.id}` },
-                });
                 await modLogsChannel.send({ embeds: [userBannedIncompleteEmbed] });
```
Comment (NathanInnes) [inline src/handlers/ban.ts:92]: return instead of await

File: src/commands/moderation/ban.ts (hunk 2 above)
Comment (benw202) [reply src/commands/moderation/ban.ts:106]: An actual form, URL will be stored with github secrets. There is also a webhook integration in a channel which drops a embed when a form is submitted

File: src/commands/moderation/ban.ts (hunk 1 above)
Comment (benw202) [reply src/commands/moderation/ban.ts:24]: Can't remember, it was there before. I haven't tested with/without yet, but there was a reason for it at the time haha

File: src/handlers/ban.ts (hunk 3 above)
Comment (benw202) [reply src/handlers/ban.ts:92]: It doesn't like a return here. (ESLint: Async arrow function 'executor' expected no return value. (consistent-return))

File: src/handlers/ban.ts [hunk 4]
```diff
@@ -12,82 +12,100 @@ module.exports = {
...
+        const noLogEmbed = makeEmbed({
+            color: Colors.Red,
+            author: {
+                name: `[BANNED] ${msg.user.tag}`,
+                iconURL: msg.user.displayAvatarURL(),
+            },
+            description: makeLines([
+                `${msg.user.tag} was banned from ${msg.guild.name} but no audit log could be found.`,
+                '',
+                '**NOTE - This was a non bot ban.**',
```
Comment (NathanInnes) [inline src/handlers/ban.ts:37]: Prefer
```suggestion
                bold('This was a non bot ban.'),
```
or the way it is?

**Comment by benw202** [comment]
Ready for review but please do not merge as waiting on other integrations/co-ordinated roll out.

**Review by NathanInnes — COMMENTED** [review]
Previous review you have only changed one line but it affects others also.

Please add the env variable to the .env.example as well :)

You may also wish to document what it's for in the contributing guide since other tokens are explained?

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -95,6 +95,9 @@ If your command requires environment variables please add them to the ```.env.ex
 2. You can do this manually or using the command line by typing ```cp .env.example .env```.
 3. Fill out the ```.env``` file with your environment variables.
 
+### Ban Appeal Form
```
Comment (NathanInnes) [inline .github/CONTRIBUTING.md:98]: Position below Tokens title would make more sense

**Review by oim1 — APPROVED** [review]

**Pull request merged**